tests/gold_tests/pluginTest/cppapi/cppapi.test.py

```diff
@@ -18,9 +18,17 @@
 Execute plugin with cppapi tests.
 '''
 
-ts = Test.MakeATSProcess("ts")
+ts = Test.MakeATSProcess("ts", command="tsx")
 
-ts.Disk.plugin_config.AddLine('test_cppapi.so')
+Test.PreparePlugin(Test.Variables.AtsTestToolsDir + '/plugins/test_cppapi.cc', ts)
+
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = (
+    'echo "#!/bin/bash" > $ATS_BIN/tsx ; ' +
+    'echo "LD_PRELOAD=$ATS_BIN/../lib/libtscppapi.so $ATS_BIN/traffic_server" >> $ATS_BIN/tsx ; ' +
+    'echo  chmod +x $ATS_BIN/tsx'
+)
+tr.Processes.Default.ReturnCode = 0
```

Comment:
I had to do something very very ugly hear to get libtscpapi.so to link dynamically.  This was not an issue when the test_cppapi plugin was under the plugins directory.

@dragon512 can you please take a look and see if there is a better way to do this?